linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -4845,27 +5007,40 @@ private[optimizer] object OptimizerCore {
     val Empty: OptEnv = new OptEnv(None, Map.empty, Map.empty)
   }
 
-  private class Scope(val env: OptEnv,
-      val implsBeingInlined: Set[Scope.InliningID]) {
-    def withEnv(env: OptEnv): Scope =
-      new Scope(env, implsBeingInlined)
+  private class Scope private (
+    val env: OptEnv,
+    val implsBeingInlined: Set[Scope.InliningID],
+    val importReplacement: Option[ImportReplacement]
```

Comment:
I'm not sure the Scope (as opposed to the Env) is the right place to put this.

My understanding at some point was:
- `OptEnv`: IR environment *before* optimizer transformations.
- `Scope`: JavaScript scope that is being constructed.

Notably, it seems the scope travels across Closure boundaries, wheres the `OptEnv` does not.

Is this understanding correct? It somewhat feels I should be able to re-use the `Binding` infrastructure for this (also: now that I think more about this: JS scopes are not really meaningful at this level, since all references we produce must be valid in IR scope again).

Reply:
> Notably, it seems the scope travels across Closure boundaries, wheres the `OptEnv` does not.

Yes, that is correct. There's no real distinction beyond that. Currently it's used to track the set of methods being inlined at the current point, so that we never inline a method within itself, even inside a `Closure` (because the method body could contain a `Closure` whose body calls the method; we must not inline in those situations).

If the information you're adding should cross `Closure` boundaries, it should go in `Scope`. Otherwise, it should go in `OptEnv`.